tinyagentos/projects/task_store.py

```
@@ -0,0 +1,291 @@
+            "DELETE FROM task_relationships WHERE id = ?", (relationship_id,)
+        )
+        await self._db.commit()
+
+    async def list_relationships(
+        self,
+        task_id: str,
+        direction: str = "from",
+    ) -> list[dict]:
+        col = "from_task_id" if direction == "from" else "to_task_id"
+        async with self._db.execute(
+            f"SELECT * FROM task_relationships WHERE {col} = ? ORDER BY created_at ASC",
```

Ruff (0.15.11)

[error] Possible SQL injection vector through string-based query construction

(S608)